harness/determined/profiler.py

```diff
@@ -101,16 +107,13 @@ def __init__(
         # If the ProfilingAgent is disabled, don't waste resources by creating useless threads
         if self.is_enabled:
             # Set up timer thread to stop collecting after 5 minutes
-            self.max_collection_seconds = 300
-            self.shutdown_timer = PreemptibleTimer(
-                self.max_collection_seconds, self._end_collection
-            )
+            self.shutdown_timer = PreemptibleTimer(MAX_COLLECTION_SECONDS, self._end_collection)
 
             # Set up the thread responsible for making API calls
             self.send_queue = (
                 queue.Queue()
             )  # type: """queue.Queue[Union[List[TrialProfilerMetricsBatch], ShutdownMessage]]"""
-            self.sender_thread = ProfilerSenderThread(self.send_queue, self.master_url)
+            self.sender_thread = ProfilerSenderThread(self.send_queue, self.master_url, 2)
```

Comment:
Won't `sender_thread` shutdown hang forever on all GPUs except for `global_rank=0` since one or both of the `sys_metric_collector_thread` and `timings_batcher_thread` won't exist?

Reply:
That wasn't quite accurate. It will hang for all GPUs that are `local_rank=0`, but `global_rank != 0` because the timings batcher thread won't exist and so the sender thread will never shut down. All other GPUs will work correcty because they are either `global_rank=0` which this code was written for or they are not `local_rank=0` in which case `is_enabled` will be `False`.

Reply:
yeh, I'll try to handle this end case. I'd rather this than drop partial batches because I think then we won't end up actually respecting start on / end after batch.

Reply:
fixed